waku2-vs-waku1/waku2-vs-waku1.md

```diff
@@ -0,0 +1,99 @@
+
+For this reason, the following approach applies to all scenarios:
+1. Simulations are run **locally**. This limits the size of possible scenarios due to local resource constraints, but is a way to quickly get an approximate comparison.
+2. Nodes are treated as a **blackbox** for which we only measure bandwidth, using an external bandwidth monitoring tool. In other words, we do not consider differences in the size of the envelope (for v1) or the message (for v2).
+3. Messages are published at a rate of **50 new messages per second** to each network, except where explicitly stated otherwise.
+4. Each message propagated in the network carries **8 bytes** of random payload, which is **encrypted**. The same symmetric key cryptographic algorithm (with the same keys) are used in both Waku v1 and v2.
+5. Traffic in each network is **generated from 10 nodes** (randomly-selected) and published in a round-robin fashion to **10 topics** (content topics for Waku v2). In practice, we found no significant difference in _average_ bandwidth usage when tweaking these two parameters (the number of traffic generating nodes and the number of topics).
+6. Peers are connected in a decentralized **full mesh topology**, i.e. each peer is connected to every other peer in the network. Waku v1 is expected to flood all messages across all existing connections. Waku v2 gossipsub will GRAFT some of these connections for full-message peerings, with the rest being gossip-only peerings.
+7. After running each scenario, we **verify that messages propagated to all peers** (comparing the number of published messages to the metrics logged by each peer).
+
+For Waku v1, nodes are configured as "full" nodes (i.e. with full bloom filter), while Waku v2 nodes are `relay` nodes, all subscribing and publishing to the same PubSub topic.
+
+## Network size comparison
+
+### Scenario 1: 10 nodes
+
+Let's start with a small network of 10 nodes only and see how Waku v1 bandwidth usage compares to that of Waku v2. At this small scale we don't expect to see improved bandwidth usage in Waku v2 over Waku v1, since all connections, for both Waku v1 and Waku v2, will be full-message connections. The number of connections is low enough that Waku v2 nodes will likely GRAFT all connections to full-message peerings, essentially flooding every message on every connection in a similar fashion to Waku v1. If our expectations are confirmed, it helps validate our methodology, showing that it gives more or less equivalent results between Waku v1 and Waku v2 networks.
+
+![](https://i.imgur.com/qffZkrn.png)
```

Comment:
I guess time indicates the elapsed time, right?

Reply:
Correct, yes!

Reply:
@jm-clius 
Wondering why the elapsed time should matter in the bandwidth usage? in most figures, the bandwidth consumption is the same regardless of elapsed time (time seems to have no influence). I would expect the number of the participating nodes or the messaging rate would matter, but could not understand the time aspect (maybe I am missing something), can you please elaborate on it.  🙏

Reply:
Good question, and I agree that it's not a very useful dimension, other than it showing the "real", repeatable test results as they were recorded.
Another thing is that they show in tests of increasing scale that it _is_ indeed only the size of the network and message rate that matters (i.e. there are no spikes due to control messages, the network reorganising, etc.). Since there are subtle differences in what the actual achieved message rate was between different scenarios, I wanted to include the "raw" test results as part of the discussion before providing the more interesting summary view that shows the difference between differently-sized networks without considering elapsed time.

Reply:
Thanks, makes sense :)